sdk/src/azure/platform/az_posix.c

```diff
@@ -26,3 +30,129 @@ AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds)
   (void)usleep((useconds_t)milliseconds * _az_TIME_MICROSECONDS_PER_MILLISECOND);
   return AZ_OK;
 }
+
+void az_platform_critical_error()
+{
+  while (1)
+  {
+  }
+}
+
+AZ_NODISCARD az_result az_platform_get_random(int32_t* out_random)
+{
+  _az_PRECONDITION_NOT_NULL(out_random);
+  
+  *out_random = (int32_t)random();
+  return AZ_OK;
+}
+
+void _thread_handler(union sigval sv)
+{
+  az_platform_timer* timer_handle = sv.sival_ptr;
+
+  _az_PRECONDITION_NOT_NULL(timer_handle);
+  _az_PRECONDITION_NOT_NULL(timer_handle->_internal.callback);
+  timer_handle->_internal.callback(timer_handle->_internal.sdk_data);
+}
+
+AZ_NODISCARD az_result az_platform_timer_create(
+    az_platform_timer* timer_handle,
+    az_platform_timer_callback callback,
+    void* sdk_data)
+{
+  _az_PRECONDITION_NOT_NULL(timer_handle);
+  memset(timer_handle, 0, sizeof(az_platform_timer));
+
+  timer_handle->_internal.callback = callback;
+  timer_handle->_internal.sdk_data = sdk_data;
+
+  timer_handle->_internal.sev.sigev_notify = SIGEV_THREAD;
+  timer_handle->_internal.sev.sigev_notify_function = &_thread_handler;
+  timer_handle->_internal.sev.sigev_value.sival_ptr = timer_handle;
+
+  if (0 != timer_create(
+          CLOCK_REALTIME, &timer_handle->_internal.sev, &timer_handle->_internal.timerid))
```

Comment:
Nit: we should decide on usage of CLOCK_REALTIME vs other types.
Looks like REALTIME is affected by a NTP change and I'm not sure we want to implement things like delay through something that is externally controlled by the network. (Maybe CLOCK_MONOTONIC is a better choice?)

Reply:
I agree, given that the platform functions are to be used by the user, I don't see a problem with it not updating with NTP. Users can always create a timer in their application with CLOCK_REALTIME. Changed!